File: Problem Solving/Super_Reduced_String/SuperReducedString.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

public class SuperReducedString {


    // Complete the superReducedString function below.
    static String superReducedString(String s) {
        for( int i=0; i<(s.length()-1); i++){

            if((s.charAt(i)==s.charAt(i+1))){
                s = i==0? s.substring(2): (i+2) < s.length()?   s.substring(0, i)+s.substring(i+2):s.substring(0, i);               
                i=-1;
            }
        }

    return s.isEmpty()?"Empty String" : s;

    }

    private static final Scanner scan = new Scanner(System.in);

    public static void main(String[] args) throws IOException {
        // BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        // BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        String s = scan.nextLine();

        String result = superReducedString(s);
        System.out.println(result);
    }
}

File: Problem Solving/Super_Reduced_String/SuperReducedString.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the superReducedString function below.
def superReducedString(s):
    for (i,x) in enumerate(s):
        if(s[i]==s[i+1] and i<len(s)):
            if(i==0):
                s=s[2:]
            elif(i+2 < len(s)):
                s=s[0:i]+s[i+2:]
            else:
                s=s[0:i]

    return "Empty String" if s=="" else s



if __name__ == '__main__':
    # fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = superReducedString(s)

    # fptr.write(result + '\n')

    # fptr.close()
